makecrx.sh

```diff
@@ -115,8 +115,19 @@ crmagic_hex="4372 3234" # Cr24
 version_hex="0200 0000" # 2
 pub_len_hex=$(byte_swap $(printf '%08x\n' $(ls -l "$pub" | awk '{print $5}')))
 sig_len_hex=$(byte_swap $(printf '%08x\n' $(ls -l "$sig" | awk '{print $5}')))
+
+# Case-insensitive matching is a GNU extension unavailable when using BSD sed.
+if [[ "$(sed --version 2>&1)" =~ "GNU" ]]; then
+  sed="sed"
+elif [[ "$(gsed --version 2>&1)" =~ "GNU" ]]; then
+  sed="gsed"
+else
+  echo "Cannot find a GNU sed. Please install one!"
```

Comment:
Thanks for the review.

It's kinda complex on macOS and with Homebrew, which is the only package manager supported by that script. By default `install-dev-dependencies` installs `gnu-sed` which `brew` then installs as `gsed`, so for most people on macOS the GNU `sed` will be called that, but `brew` also offers a `--with-default-names` option to install GNU `sed` as just `sed`. Consequently there's some ambiguity as to what it is likely to be called.

We could trim the `echo` out here & just assume people are running that dependencies script, which seems reasonable, but I think the naming logic is probably something that needs to remain in some shape or form.

@J0WI was probably right that ideally this could be handled by something like `perl` which would be more cross-platform with less risk in terms of unexpected variability, but my `perl` knowledge is pretty awful sadly.